src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -878,7 +878,30 @@ private static bool IsGlobalAttributeTarget(SyntaxToken token)
 
         private bool IsPossibleAttributeDeclaration()
         {
-            return this.CurrentToken.Kind == SyntaxKind.OpenBracketToken;
+            // Have to at least start with `[` to be an attribute
+            if (this.CurrentToken.Kind != SyntaxKind.OpenBracketToken)
+                return false;
+
+            using var _ = this.GetDisposableResetPoint(resetOnDispose: true);
+
+            // Eat the `[`
+            EatToken();
```

Comment:
>EatToken(); [](http://example.com/codeflow?start=12&length=11)

Is it normal for one of the Is* methods to modify state by eating a token?

Reply:
100%.  THat's why there is a resetpoint.  You can speculatively parse to your heart's content (as long as you return back to the original parser state when the functino exits).